src/main/java/uk/gov/hmcts/reform/preapi/tasks/CleanupLiveEvents.java

```diff
@@ -1,39 +1,201 @@
+                      .getResourceState().equals(LiveEventResourceState.RUNNING.toString())
+                  )
+                  .forEach(liveEventDTO -> {
+                      log.info("Finding capture session by live event id {}", liveEventDTO.getName());
+                      var sendNotification = false;
+                      try {
+                          var captureSession = captureSessionService.findByLiveEventId(liveEventDTO.getName());
+                          var search = new SearchRecordings();
+                          log.info("Finding recordings by capture session {}", captureSession.getId());
+                          search.setCaptureSessionId(captureSession.getId());
+                          var recordings = recordingService.findAll(search, false, Pageable.unpaged());
+
+                          if (recordings.isEmpty()) {
+                              log.info("No recordings found for capture session {}", captureSession.getId());
+                              sendNotification = stopLiveEvent(
+                                  mediaService,
+                                  captureSession,
+                                  UUID.randomUUID(),
+                                  liveEventDTO.getName()
+                              );
+                          } else {
+                              sendNotification = !recordings.map(recording -> {
+                                  log.info(
+                                      "{} recordings found for capture session {}",
+                                      recordings.getSize(),
+                                      captureSession.getId()
+                                  );
+                                  return stopLiveEvent(
+                                      mediaService,
+                                      captureSession,
+                                      recording.getId(),
+                                      liveEventDTO.getName()
+                                  );
+                              })
+                                                            .filter(b -> b)
+                                                            .toList()
+                                                            .isEmpty();
+                          }
+
+                          if (sendNotification) {
+                              var booking = bookingService.findById(captureSession.getBookingId());
+                              if (booking != null) {
+                                  var toNotify = booking.getShares().stream()
+                                                        .map(shareBooking -> userService.findById(
+                                                            shareBooking.getSharedWithUser().getId())
+                                                        )
+                                                        .filter(u -> u != null && u.getDeletedAt() == null)
```

Comment:
User service will throw not found error if cannot find user (so it will never be null)
```suggestion
                                                        .filter(u -> u.getDeletedAt() == null)
```